src/wp-includes/rest-api/endpoints/class-wp-rest-site-health-controller.php

```diff
@@ -0,0 +1,244 @@
+<?php
+/**
+ * REST API: WP_REST_Site_Health_Controller class
+ *
+ * @package WordPress
+ * @subpackage REST_API
+ * @since 5.5.0
+ */
+
+/**
+ * Core class for interacting with Site Health elements.
+ *
+ * @since 5.5.0
+ *
+ * @see WP_REST_Controller
+ */
+class WP_REST_Site_Health_Controller extends WP_REST_Controller {
+
+	private $site_health;
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		if ( ! class_exists( 'WP_Site_Health' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/class-wp-site-health.php';
+		}
+
+		$this->namespace = 'wp-site-health/v1';
+
+		$this->site_health = WP_Site_Health::get_instance();
+	}
+
+	/**
+	 * Register API routes.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @see register_rest_route()
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			sprintf(
+				'/%s/%s',
+				'test',
```

Comment:
I think all these `test` should be `tests`. We can also set it as the `rest_base` in the controller. That would match our pattern of using a plural collection base.